cmd/query/app/server.go

```diff
@@ -79,14 +81,35 @@ func createHTTPServer(querySvc *querysvc.QueryService, queryOpts *QueryOptions,
 
 	apiHandler.RegisterRoutes(r)
 	RegisterStaticHandler(r, logger, queryOpts)
-	compressHandler := handlers.CompressHandler(r)
+	var compressHandler http.Handler
+	if queryOpts.BearerTokenPropagation {
+		compressHandler = handlers.CompressHandler(BearTokenPropagationHandler(logger, r))
+	} else {
+		compressHandler = handlers.CompressHandler(r)
+	}
 	recoveryHandler := recoveryhandler.NewRecoveryHandler(logger, true)
-
 	return &http.Server{
 		Handler: recoveryHandler(compressHandler),
 	}
 }
 
+func BearTokenPropagationHandler(logger *zap.Logger, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		authHeaderValue := r.Header.Get("Authorization")
```

Comment:
What is the format of Authorization header?

Reply:
According to https://tools.ietf.org/html/rfc6750#section-2.1
It should be in the form : Authorization: "Bearer <token>"